### PR TITLE
ac-34 api documentation for people api

### PR DESCRIPTION
**Pull request by ExtraGravity**

### CODE REVIEW

File: src/routers/people.ts [hunk 1]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+ *
+ * @apiError (401) Unauthorized - Must be admin
+ *
+ * @apiSuccess (200) {String} Successfully retrieved all people
+ * @apiSuccessExample Success-Response:
+  [
+    {
+      "personUuid": "1e167b81-d816-497b-8c0c-36f4d6b2fd33",
+      "category": "farmers",
+      "firstName": "Zachariah",
```
Comment (beaesguerra) [inline src/routers/people.ts:29]: don't need middleName  and lastName?

File: src/routers/people.ts [hunk 2]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+    }
+  ]
+ */
 router.get('/', async (req, res) => {
   res.status(StatusCode.OK).send('Successfully retrieved all people');
 }, authorized(UserType.ADMIN));
 
+/**
+ * @api {get} /people/<category>
+ * @description Returns all people and all their associated attributes.
```
Comment (beaesguerra) [inline src/routers/people.ts]: update to "Returns all people of this category type and all their associated attributes"?

File: src/routers/people.ts [hunk 3]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+ *              Parameters have no effect on this request, but the URL
+ *              specifies what kind of people are given.
+ *              Parameters have no effect on this request.
```
Comment (beaesguerra) [inline src/routers/people.ts]: this line is repeated

File: src/routers/people.ts [hunk 4]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+ *              Associated attributes can be checked via /peopleCategories
+ * @apiName GetPeopleCategory
+ *
+ * @apiError (401) Unauthorized - Must be admin
```
Comment (beaesguerra) [inline src/routers/people.ts]: fix this comment ple

File: src/routers/people.ts [hunk 5]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+      "personUuid": "4b3f23a3-04c4-468f-bdf5-f189a34d9f69",
+      "firstName": "Zachariah",
+      "paymentFrequency": "monthly",
+      "notes": "Has a nice car",
+    }
+  ]
+ */
 router.get('/:category', async (req, res) => {
-  res.status(StatusCode.OK).send('Successfully retrieved all people of type <something>');
+  res.status(StatusCode.OK).send('Successfully retrieved all people of category <something>');
```
Comment (beaesguerra) [inline src/routers/people.ts:76]: can we include the category that was inputted in the response message?

File: src/routers/people.ts [hunk 6]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
 }, authorized(UserType.ADMIN));
 
+/**
+ * @api {get} /people/<category>
```
Comment (beaesguerra) [inline src/routers/people.ts]: this is a post

File: src/routers/people.ts [hunk 7]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+ * @description Creates a new person in the specified category.
+ *              Returns the UUID created for that person.
+ * @apiName CreatePerson
```
Comment (beaesguerra) [inline src/routers/people.ts]: in these docs, don't we need to specify what to include in the request? ie. the parameters? like firstName, middleName, lastName, etc?

File: src/routers/people.ts [hunk 8]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+ * @apiError (401) Unauthorized - Must be admin
+ * @apiError (404) Person not found in category <category>
+ *
+ * @apiSuccess (200) {String} Successfully retrieved person of category <category>
+ * @apiSuccessExample /people/farmers Success-Response:
+ * { personUuid: "1e167b81-d816-497b-8c0c-36f4d6b2fd33" }
+ */
 router.post('/:category', async (req, res) => {
-  res.status(StatusCode.CREATED).send('Successfully created new <type>');
+  res.status(StatusCode.OK).send('Successfully created new <category>');
```
Comment (beaesguerra) [inline src/routers/people.ts]: why aren't we using CREATED anymore? can we also add the category type in the response?

File: src/routers/people.ts [hunk 9]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
 }, authorized(UserType.ADMIN));
 
+/**
+ * @api {get} /people/<category>/<uuid>
+ * @description Gets a particular person
```
Comment (beaesguerra) [inline src/routers/people.ts]: specify that you don't need params?

File: src/routers/people.ts [hunk 10]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+   }
+ */
 router.get('/:category/:uuid', async (req, res) => {
-  res.status(StatusCode.OK).send('Successfully got <type>');
+  res.status(StatusCode.OK).send('Successfully got <category>');
 }, authorized(UserType.ADMIN));
 
+/**
+ * @api {put} /people/<category>/<uuid>
+ * @description Returns a specific person from a specific category
```
Comment (beaesguerra) [inline src/routers/people.ts]: this should say it updates? not returns?

File: src/routers/people.ts [hunk 11]
```diff
@@ -8,24 +8,127 @@ import { StatusCode } from '@/models/statusCodes';
...
+ *              and all their associated attributes.
+ *              Parameters have no effect on this request.
+ *              Associated attributes can be checked via /peopleCategories
+ * @apiName UpdatePerson
+ *
+ * @apiParam {String} [<attribute>] An attribute of a person with its value.
```
Comment (beaesguerra) [inline src/routers/people.ts]: maybe mention in the apiParams to check the /peopeCategories api to see all the possible attributes

File: src/routers/peopleCategories.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+const router = createRouter();
+
+/**
+ * @api {get} /peopleCategories
+ * @description Returns all categories of people and their associated attributes
+ *
+ * @apiName GetPeopleCategories
+ *
+ * @apiError (401) Unauthorized - Must be admin
```
Comment (beaesguerra) [inline src/routers/peopleCategories.ts]: does not have to be admin here

**Review by beaesguerra — CHANGES_REQUESTED** [review]
pls fix what I've commented on, and when you're done, I'll review the autogenerated documentation the next time

File: src/routers/people.ts (hunk 1 above)
Comment (ExtraGravity) [reply src/routers/people.ts:29]: The API doesn't need to guarantee it. It's something that is checkable in `/peopleCategories`. I wanted to keep the docs shorter so you weren't scrolling forever.

File: src/routers/people.ts (hunk 4 above)
Comment (ExtraGravity) [reply src/routers/people.ts]: you fix it

File: src/routers/people.ts (hunk 5 above)
Comment (ExtraGravity) [reply src/routers/people.ts:76]: I did

File: src/routers/people.ts (hunk 11 above)
Comment (ExtraGravity) [reply src/routers/people.ts]: Was mentioned in description

File: src/routers/people.ts (hunk 9 above)
Comment (ExtraGravity) [reply src/routers/people.ts]: Done

File: src/routers/people.ts [hunk 13]
```diff
@@ -74,28 +76,30 @@ router.get('/:category', async (req, res) => {
...
+ * @api {post} /people/<category>
  * @description Creates a new person in the specified category.
  *              Returns the UUID created for that person.
+ *              Requires all associated attributes given in /peopleCategories
  * @apiName CreatePerson
  *
- * @apiError (401) Unauthorized - Must be admin
- * @apiError (404) Person not found in category <category>
+ * @apiError (400) Missing or invalid fields, ...
+ * @apiError (401) Unauthorized
```
Comment (beaesguerra) [inline src/routers/people.ts]: can u add the @apiParam tag and mention to check the categories api? maybe do this for all of them :D

File: src/routers/people.ts (hunk 11 above)
Comment (beaesguerra) [reply src/routers/people.ts]: Can you move that comment to the apiParam tag? :)

**Review by beaesguerra — APPROVED** [review]

**Pull request merged**